src/Symfony/Bundle/FrameworkBundle/Client.php

```diff
@@ -104,6 +106,41 @@ public function enableReboot()
         $this->reboot = true;
     }
 
+    /**
+     * Logins a user against a specific firewall.
+     *
+     * @param string      $username
+     * @param string      $password
+     * @param string[]    $roles
+     * @param string      $firewallName
+     * @param string|null $firewallContext
+     */
+    public function login(
+        string $username,
+        string $password,
+        array $roles = array('ROLE_ADMIN'),
+        string $firewallName = 'main',
+        string $firewallContext = null
+    ) {
+        $context = $firewallContext ?? $firewallName;
+        $session = $this->getContainer()->get('session');
+
+        $token = new UsernamePasswordToken($username, $password, $firewallName, $roles);
```

Comment:
FrameworkBundle can be used without security (hidden by the fact it has `security-*` as dev requirements which keeps tests green). 
Should this helper go in SecurityBundle instead? Perhaps as a trait?

Reply:
of throw at the beginning of this method if you try to use this helper without the security component (you don't need the helper). Might be much easier for DX than asking people to import a trait in their test cases.

Reply:
thus, the current architecture puts it on the Client, not as a testcase helper, making it even harder to move it elsewhere.